app/questionnaire/questionnaire_schema.py

```diff
@@ -82,7 +82,8 @@ def __init__(
         ] = defaultdict(set)
         self._language_code = language_code
         self._questionnaire_json = questionnaire_json
-        self._repeating_block_ids: list[str] = []
+        self._repeating_block_to_list_map: dict[str, str] = {}
```

Comment:
I think we need to prefix repeating block with `list` otherwise it might get confusing as we have repeating sections, and technically all blocks inside that is a repeating block, but we are specifically talking about repeating blocks for a list. Thoughts?

Reply:
Yeah that makes sense, I'll change it to `_list_names_by_list_repeating_block` to incorporate the list and keep it similar to other dict names